internal/buildstamp/buildstamp.go

```diff
@@ -129,3 +132,35 @@ func (v Vars) String() string {
 	fmt.Fprintf(&sb, "clean build: %v\n", v.IsClean)
 	return sb.String()
 }
+
+type ErrUnofficialBuild struct {
+	parent *Vars
+}
+
+func (e ErrUnofficialBuild) Error() string {
+	why := "unknown reason"
+	if !e.parent.IsClean {
+		why = "built with modified source files"
+	} else if !gitOfficialBranchRe.MatchString(e.parent.SourceBranch) {
+		why = "built from non-release branch"
+	}
+	return "build is not official: " + why
+}
+
+func (v Vars) GetVersion() (string, error) {
+	if v == emptyValues {
+		return "", ErrStampingDisabled
+	}
+	if !v.IsOfficial || v.ReleaseVersion == unknown {
+		return fmt.Sprintf("v0.0.0-%s-%s", compactTime(v.BuildTimestamp), shortHash(v.SourceRevision)), nil
```

Comment:
Do you think it'd be offensive to the existing format to suffix with a `+dirty`? I didn't see an appropriate place to put this info in the Go module pseudo-version format.

It would only be relevant to us - but maybe a clue in our dev clusters that people are trying to auth with random builds of `engflow_auth` that may not even be built from clean source (i.e. OK for oncall to ignore)

Reply:
Seems reasonable. It's not a valid Go pseudo-version (`+incompatible` is the only allowed build tag), but it is a valid semantic version, and that's what matters for this purpose.